tests/unit/amo/pages/TestHome-new.js

```diff
@@ -317,4 +338,496 @@ describe(__filename, () => {
       });
     });
   });
+
+  describe('Tests for HeroRecommendation', () => {
+    describe('for an addon', () => {
+      it('renders a heading', () => {
+        const name = 'Addon name';
+        renderWithHomeData({
+          primaryProps: {
+            addon: { ...fakeAddon, name: createLocalizedString(name) },
+          },
+        });
+
+        expect(screen.getByRole('heading', { name })).toBeInTheDocument();
+      });
+
+      it('renders a link', () => {
+        const slug = 'some-addon-slug';
+        renderWithHomeData({
+          primaryProps: {
+            addon: { ...fakeAddon, slug },
+          },
+        });
+
+        expect(
+          screen.getByRole('link', { name: 'Get the extension' }),
+        ).toHaveAttribute(
+          'href',
+          addQueryParams(`/${lang}/${clientApp}/addon/${slug}/`, {
+            utm_source: DEFAULT_UTM_SOURCE,
+            utm_medium: DEFAULT_UTM_MEDIUM,
+            utm_content: PRIMARY_HERO_SRC,
+          }),
+        );
+      });
+
+      it.each([
+        [LINE, 'BY FIREFOX'],
+        [RECOMMENDED, 'RECOMMENDED'],
+        [SPONSORED, 'SPONSORED'],
+        [VERIFIED, 'SPONSORED'],
+        ['unknown category', 'SPONSORED'],
+      ])('displays the expected title for %s add-ons', (category, title) => {
+        renderWithHomeData({
+          primaryProps: {
+            addon: addonForPromotedCategory(category),
+          },
+        });
+
+        expect(screen.getByText(title)).toHaveClass(
+          'HeroRecommendation-title-text',
+        );
+      });
+
+      it.each([SPONSORED, VERIFIED, 'unknown category'])(
```

Comment:
interesting that we treat an unknown category as SPONSORED/VERIFIED.  Is that what we want?

(I'm not necessarily saying we should change anything, just mentioning it)

Reply:
I think it was future-proofing in case some new category was added. Seeing as we never use any of these categories now, other then recommended, it's kind of irrelevant, but I think the code is ok.

Reply:
Fair enough, but I don't really want to be changing anything about any component's behaviour in any of these patches, unless absolutely necessary.